.github/workflows/publish_pkgci_mi300.yml

```diff
@@ -0,0 +1,44 @@
+name: Publish manylinux_x86_64 image
+on:
+  workflow_dispatch:
+  push:
+    branches: ['main']
+    paths:
+    - dockerfiles/pkgci_mi300.Dockerfile
+    - .github/workflows/publish_pkgci_mi300.yml
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: iree-org/pkgci_mi300
```

Comment:
(update names and paths after addressing the other comment about the dockerfile name)